e2e-tests/src/test/scala/pl/touk/nussknacker/BatchDataGenerationSpec.scala

```diff
@@ -0,0 +1,74 @@
+package pl.touk.nussknacker
+
+import io.circe.syntax.EncoderOps
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.test.{NuRestAssureExtensions, NuRestAssureMatchers, VeryPatientScalaFutures}
+import pl.touk.nussknacker.ui.process.marshall.CanonicalProcessConverter.toScenarioGraph
+import io.circe.parser._
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import scala.util.Try
+
+class BatchDataGenerationSpec
+    extends AnyFreeSpecLike
+    with DockerBasedInstallationExampleNuEnvironment
+    with Matchers
+    with VeryPatientScalaFutures
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers {
+
+  "Batch scenario generate file function should generate random results according to defined schema" in {
+    given()
+      .when()
+      .request()
+      .preemptiveBasicAuth("admin", "admin")
+      .jsonBody(s"""
+          |{
+          |    "name" : "SumTransactions",
+          |    "category" : "Default",
+          |    "isFragment" : false,
+          |    "processingMode" : "Bounded-Stream"
+          |}
+          |""".stripMargin)
+      .post("http://localhost:8080/api/processes")
+      .Then()
+      .statusCode(201)
+
+    val testResultContent = given()
```

Comment:
The same goes for the "when" block. The extracting part should be also encapsulated in there.

Reply:
I changed it to use a response body matcher extension